server/src/test/java/io/spine/server/event/EventRootCommandIdShould.java

```diff
@@ -0,0 +1,192 @@
+        assertEquals(command.getId(), getRootCommandId(events.get(0)));
+        assertEquals(command.getId(), getRootCommandId(events.get(1)));
+        assertEquals(command.getId(), getRootCommandId(events.get(2)));
+    }
+
+    @Test
+    public void match_the_id_of_an_external_event_handled_by_an_aggregate() {
+        final Command command = command(createProject(projectId(), teamId()));
+
+        postCommand(command);
+
+        final List<Event> events = readEvents();
+        // Two events should be created: one in the `ProjectAggregate` and one in the `TeamAggregate`.
+        assertSize(2, events);
+
+        final Event reaction = events.get(1);
+        assertEquals(command.getId(), getRootCommandId(reaction));
+    }
+
+    @Test
+    public void match_the_id_of_a_command_handled_by_a_process_manager() {
+        final Command command = command(addTeamMember(teamId()));
+
+        postCommand(command);
+
+        final List<Event> events = readEvents();
+        assertSize(1, events);
+
+        final Event event = events.get(0);
+        assertEquals(command.getId(), getRootCommandId(event));
+    }
+
+    @Test
+    public void match_the_id_of_a_command_handled_by_a_process_manager_for_multiple_events() {
+        final Command command = command(inviteTeamMembers(teamId(), 3));
+
+        postCommand(command);
+
+        final List<Event> events = readEvents();
+        assertSize(3, events);
+        assertEquals(command.getId(), getRootCommandId(events.get(0)));
+        assertEquals(command.getId(), getRootCommandId(events.get(1)));
+        assertEquals(command.getId(), getRootCommandId(events.get(2)));
+    }
+
+    @Test
+    public void match_the_id_of_an_external_event_handled_by_a_process_manager() {
+        final Command command = command(acceptInvitation(teamId()));
+
+        postCommand(command);
+
+        final List<Event> events = readEvents();
+        // Two events should be created: the `InvitationAccepted` and the `TeamMemberAdded`.
+        assertSize(2, events);
+
+        final Event reaction = events.get(1);
+        assertEquals(command.getId(), getRootCommandId(reaction));
+    }
+
+    private static Command command(Message message) {
+        return requestFactory.createCommand(message);
+    }
+
+    private void postCommand(Command command) {
+        final StreamObserver<Ack> observer = noOpObserver();
+        boundedContext.getCommandBus()
+                      .post(command, observer);
+    }
+
+    /**
+     * Reads all events from the bounded context event store.
+     */
+    private List<Event> readEvents() {
+        final EventStreamQuery query = allEventsQuery();
+        final MemoizingObserver<Event> observer = StreamObservers.memoizingObserver();
+
+        boundedContext.getEventBus()
+                      .getEventStore()
+                      .read(query, observer);
+
+        final List<Event> results = observer.responses();
+        return results;
+    }
+
```

Comment:
Please fix all the similar cases. If you look through the codebase, you'll notice we don't allow extra empty lines.